components/Typeahead.tsx

```diff
@@ -9,17 +9,19 @@ import 'react-bootstrap-typeahead/css/Typeahead.css'; //Typeahead styling
 import Helmet from 'react-helmet'; // Helmet
 import 'bootstrap/dist/css/bootstrap.min.css'; //Bootstrap styling
 import { Center, Divider, VStack, Text } from '@chakra-ui/layout'; // Chakra UI Layout
-import { Spacer } from '@chakra-ui/react'; //Chakra UI React
+import { Spacer, Spinner } from '@chakra-ui/react'; //Chakra UI React
 
 // Typeahead props
 type Props<T extends TypeaheadModel> = Pick<
   AsyncTypeaheadProps<T>,
   | 'isLoading' // boolean to indicate if query is loading
   | 'onSearch' // function to execute query when text is entered in input field
   | 'renderMenuItemChildren' // function to format each result in the menu
+  | 'labelKey' // string used for searching and rendering typeahead results
 > & {
   results: Array<T>; // array of results to display in the typeahead menu
   placeholder: string; // placeholder text
+  setSelected: (selected: T | undefined) => void; // record selected item
```

Comment:
```suggestion
  onSelect: (selected: T | undefined) => void; // record selected item
```
Let's name this callback `onSelect` as it's more flexible (it should be able to do anything given the selected option)